src/token.rs

```diff
@@ -17,7 +17,7 @@ pub trait TokenCache: Sync {
     /// the token.
     async fn get(&self, client: &reqwest::Client) -> crate::Result<String> {
         match self.token_and_exp().await {
-            Some((token, exp)) if now() > exp => Ok(token),
+            Some((token, exp)) if now() < exp => Ok(token),
```

Comment:
I was just about to submit an issue related to this but will comment here instead.

Have you considered adding a leeway to the expiry check to avoid some subtle edge cases, e.g. we can return a token that from get that expires in 10ms but when it arrives at Google it is expired due to latency.

Adding a leeway of say 60s or 300s here might be reasonable, i.e. refreshing the token 1 or 5 minutes ahead of time.

```rs
Some((token, exp)) if now() + leeway < exp => Ok(token),
```